hivesim-rs/src/testapi.rs

```diff
@@ -417,12 +420,15 @@ async fn run_n_client_test(
 
             let mut client_vec: Vec<Client> = Vec::new();
             for (index, client) in clients.into_iter().enumerate() {
-                let private_key = if let Some(private_keys) = private_keys.clone() {
-                    private_keys[index].to_owned()
+                let environment_variable = if let Some(environment_variable) = environment.clone() {
+                    environment_variable[index].to_owned()
```

Comment:
This all feels like a hacky version of `zip()`, which forces a lot of unnecessary cloning. I know this was already here, but while we're here, we can [fix some broken windows](https://blog.codinghorror.com/the-broken-window-theory/). Obviously performance isn't of primary importance here, but it seems worth using the shorter, cleaner implementation when it makes sense.

You could try something like:
```rs
            let env_iter = environments.map(|e| e.into_iter()).ok_or(iter::repeat(None));
            for (client, environment) in clients.into_iter().zip(env_iter) {
```

Note that I think `environment_variable` is a misnomer here, because it's a mapping of many variables. I think the vec should be `environments` and the one for each client can be the singular.